src/lexer.coffee

```diff
@@ -982,16 +977,17 @@ exports.Lexer = class Lexer
     [locationData.last_line, locationData.last_column, endOffset] =
       @getLineAndColumnFromChunk offsetInChunk + lastCharacter
     [locationData.last_line_exclusive, locationData.last_column_exclusive] =
-      @getLineAndColumnFromChunk offsetInChunk + lastCharacter + 1
+      @getLineAndColumnFromChunk offsetInChunk + lastCharacter + (if length > 0 then 1 else 0)
     locationData.range[1] = if length > 0 then endOffset + 1 else endOffset
 
     locationData
 
   # Same as `token`, except this just returns the token without adding it
   # to the results.
-  makeToken: (tag, value, {offset: offsetInChunk = 0, length = value.length, origin} = {}) ->
+  makeToken: (tag, value, {offset: offsetInChunk = 0, length = value.length, origin, generated} = {}) ->
```

Comment:
This was actually "missing" in the sense that I've added some `@token()` or `@makeToken()` calls with `generated: yes` and it wasn't doing anything since this wasn't wired up to accept that option

Reply:
Why not do the same for comments?

Reply:
I don't think I understand, are you saying allow comments to be passed in as an option to the `token()`/`makeToken()` "constructor"?

Reply:
Yes, currently you need to make the token and then add the comments afterward as a property. That used to be the case for `generated` but it isn’t now because of this change, I was asking if it makes sense to make the same change for comments.

Reply:
That would make sense to me as a pattern

Looking at the implementation of `commentToken()`, it looks like typically comments get attached to an existing previous token so wouldn't be available yet as an argument at the time of construction (of that previous token)